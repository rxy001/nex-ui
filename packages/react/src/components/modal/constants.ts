export const MODAL_INTERNAL_ID_PREFIX = 'nui-modal-'
