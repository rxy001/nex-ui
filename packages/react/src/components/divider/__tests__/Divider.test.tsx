import {
  testComponentStability,
  renderWithNexUIProvider,
  testRefForwarding,
  testRootClassName,
  testVariantDataAttrs,
} from '~/tests/shared'
import { Divider } from '../index'
import { dividerClasses, dividerDataAttrs } from './constants'

describe('Divider', () => {
  testComponentStability(<Divider />)

  testRefForwarding(<Divider />)

  testRootClassName(<Divider />)

  testVariantDataAttrs(<Divider />, ['orientation', ['horizontal', 'vertical']])

  it('should render with default props', () => {
    const { container } = renderWithNexUIProvider(<Divider />)
    const dividerRoot = container.firstElementChild

    expect(dividerRoot).toHaveClass(dividerClasses.root)
    expect(dividerRoot).toHaveAttribute(
      ...dividerDataAttrs['orientation-horizontal'],
    )

    expect(dividerRoot).toMatchSnapshot()
  })
})
