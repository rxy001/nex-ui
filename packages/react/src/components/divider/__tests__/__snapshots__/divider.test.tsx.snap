// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider renders correctly 1`] = `
.emotion-0 {
  margin: 0;
  border: none;
  background-color: var(--nui-colors-gray-muted);
  height: var(--nui-sizes-px);
  margin-block: var(--nui-spaces-6);
}

<hr
  class="nui-divider-root nui-divider-orientation-horizontal emotion-0"
  role="separator"
/>
`;
